contracts/farm/LiquidityMining.sol

```diff
@@ -731,4 +764,65 @@ contract LiquidityMining is ILiquidityMining, LiquidityMiningStorage {
         UserInfo memory ui = getUserInfo(_poolToken, _user);
         return ui.amount;
     }
+
+    /**
+     * @notice returns the accumulated liquid reward for the given user for each pool token
+     * @param _user the address of the user
+     */
+    function getUserAccumulatedLiquidReward(address _user)
+        external
+        view
+        returns (uint256[] memory)
+    {
+        uint256 length = poolInfoList.length;
+        uint256[] memory rewardList = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            address _poolToken = address(poolInfoList[i].poolToken);
+            uint256 _unlockedImmediatelyPercent =
+                _getPoolTokenUnlockedImmediatelyPercent(_poolToken);
+            rewardList[i] =
+                (_unlockedImmediatelyPercent * _getUserAccumulatedReward(i, _user)) /
+                10000;
+        }
+
+        return rewardList;
+    }
+
+    /**
+     * @notice returns the accumulated vested reward for the given user for each pool token
+     * @param _user the address of the user
+     */
+    function getUserAccumulatedVestedReward(address _user)
+        external
+        view
+        returns (uint256[] memory)
+    {
+        uint256 length = poolInfoList.length;
+        uint256[] memory rewardList = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            address _poolToken = address(poolInfoList[i].poolToken);
+            uint256 _unlockedImmediatelyPercent =
+                _getPoolTokenUnlockedImmediatelyPercent(_poolToken);
+            rewardList[i] =
+                ((10000 - _unlockedImmediatelyPercent) * _getUserAccumulatedReward(i, _user)) /
+                10000;
+        }
+
+        return rewardList;
+    }
+
+    /**
+     * @dev get the unlocked immediate percentage of specific pool token
+     * use the unlockedImmediatelyPercentOverwrite by default, if it is not set, then use the unlockedImmediatelyPercent
+     */
+    function _getPoolTokenUnlockedImmediatelyPercent(address _poolToken)
+        internal
+        view
+        returns (uint256)
+    {
+        return
+            unlockedImmediatelyPercentOverwrite[_poolToken] > 0
```

Comment:
reading from the storage twice.  
consider assigning to a variable.

Reply:
updated